### DISCUSSION TITLE
Obtain .jar File using Gradle for Java

### DISCUSSION THREAD

**Discussion by mciolkosz** [community/community#26016] [Actions]
<p>Is there some way to obtain the .jar file from the default Gradle for Java action? I would like to use the file that’s normally output at <code>build/libs/&lt;project&gt;.jar</code>. Didn’t see anything in the support topics on this, and I’m extremely new to GitHub Actions, so if it’s something super obvious, apologies for the basic question. Thanks.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mciolkosz">@mciolkosz</a>,</p>
<p>You can upload the jar file as an artifact, and download artifacts from the workflow.</p>
<pre><code class="lang-auto">- uses: actions/upload-artifact@v2
  with:
    name: my-artifact
    path: build/libs/project.jar
</code></pre>
<p>You can also upload the directory ‘build/libs’ to upload all jar files.</p>
<p>Please check official doc for more details:<br>
<a href="https://docs.github.com/en/actions/language-and-framework-guides/building-and-testing-java-with-gradle#packaging-workflow-data-as-artifacts" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/actions/language-and-framework-guides/building-and-testing-java-with-gradle#packaging-workflow-data-as-artifacts</a></p>
<p>Thanks</p>
